src/window.rs

```diff
@@ -478,7 +478,7 @@ impl WindowBuilder {
     /// - **Android / iOS / Wayland / Web:** Unsupported.
     #[inline]
     pub unsafe fn with_parent_window(mut self, parent_window: Option<RawWindowHandle>) -> Self {
```

Comment:
If `RawWindowHandle` isn't thread-safe how is this supposed to work?
I guess it's `unsafe`, so it's fine, but maybe panicking would be better here.

Actually, this might be better suited for another PR, but as pointed out above, maybe this should be moved into `PlatformSpecificWindowBuilderAttributes` and call a backend-specific function and handle it there, including panicking if we aren't on the main thread.

Reply:
You just set a data, winit can only read this data on the thread event loop is on.